Automattic/wp-calypso
# Checkout: Add fields for per-country tax requirements

Some countries require city, state, and organization to be provided in order for us to store the transaction with our tax provider. This PR provides API and UI updates.

References 1348-gh-Automattic/payments-shilling

## Proposed Changes

* Type information for the updates to the countries API
* Dynamically adds fields during checkout to support these requirements
* Stores the new data in tax location.

## Testing Instructions

* Apply D100762-code
* Run this test in store sandbox (PCYsg-IA-p2)
* Add a product (like a plan upgrade) to your cart and make sure you're on the checkout page
* Verify the previous behavior continues for countries unaffected by the parent ticket (country + postal code if relevant):
* <img width="557" alt="image" src="https://user-images.githubusercontent.com/38750/217879925-0a5f0057-ec81-4c3c-85ea-769bcb96e753.png">
* Switch between the countries in the parent ticket
* Verify that each one shows the expected fields:
* <img width="557" alt="Screen Shot 2023-02-08 at 9 58 57 AM" src="https://user-images.githubusercontent.com/38750/217880104-cb0526b9-8eb5-4daa-9ed3-00eead49c1fb.png">
* <img width="561" alt="Screen Shot 2023-02-08 at 10 00 02 AM" src="https://user-images.githubusercontent.com/38750/217880128-b64c46e6-b16b-449f-8594-7ef04484cb98.png">
* <img width="559" alt="Screen Shot 2023-02-08 at 10 01 00 AM" src="https://user-images.githubusercontent.com/38750/217880153-f0fe9092-9e9a-409d-8d8c-9fd0ae20a0ee.png">
* <img width="559" alt="Screen Shot 2023-02-08 at 10 01 40 AM" src="https://user-images.githubusercontent.com/38750/217880284-1457631e-ccdc-4e35-aa3f-212f147ed3e6.png">
* Verify that if city appears, it can't be left empty:
* <img width="560" alt="Screen Shot 2023-02-08 at 2 16 49 PM" src="https://user-images.githubusercontent.com/38750/217880364-086ab8d8-0f9c-4b93-a7df-23a63ffa555d.png">
* Add a `.ca` domain name to the cart (skip email)
* Verify the billing information is replaced with contact information
* Switch between the countries in the parent ticket
* Verify that countries that want a business name provided have the organization field automatically expanded:
* <img width="559" alt="image" src="https://user-images.githubusercontent.com/38750/217881938-06667978-f819-4d0b-90a5-a76ef3fb6912.png">
* Switch the country to "United Kingdom"
* Check "Add VAT details"
* Verify the organization field doesn't appear (it's already provided above)
